### ISSUE TITLE
Forge Bug

### ISSUE THREAD

**Issue by D3athrow-Issues** [d3athrow/vgstation13#32345]
(WEB REPORT BY: dangitmaster REMOTE: 206.221.180.138:7777)
# Revision
4990a79704d91305fd563bfbf6a105f8cd80c6d1

# Description
I made too many OP swords
# Steps to Reproduce
>get phazon hammer
>Get phazon blade unforged
>forge it
>Ding it 12 times, it won't break i know i did it three times in a row
>Get unlimited legendary phazon swords
# What you Expected
Some RNG
# What Actually Happened
UNLIMITED BLADE WORKS

**Comment by Eneocho** [comment]
Hammer quality affects how much you increase the quality value (and thus the roll for quality) of the unforged item per hit. If you were to get a regular iron hammer with quality of say 1000, each hit would increase the recipient's quality value by 1000.
Phazon parts are naturally very high quality by themselves given their rarity, so combining that with a phazon hammer (which probably was high quality itself) makes you roll very high. 
Intended. You just stacked a lot of buffs to the quality rolls.

**Comment by Exxion** [comment]
Wow, this code sucks ass
It's a runtime caused when attempting to generate a description for the resulting item, because for each point of material quality, it adds one random name to a list picked from various other sources *if that name is not already in the list*, then just assumes that it was added to the list anyway and tries to access out of bounds
It also sucks for more reasons, but that's the relevant one

**Comment by Exxion** [comment]
Wait, no, that runtime *does* happen but it's not the cause of this

**Comment by Exxion** [comment]
And I misread what the report was saying, and don't know enough about smithing to know if what it's actually saying is a bug

**Comment by WindowsErrors** [comment]
what a turnaround from the original diagnosis

**Comment by Eneocho** [comment]
Telling you two. Not a bug. He just stacked quality modifiers with a high quality hammer and both it and the sabre being made with phazon parts.

**Comment by Exxion** [comment]
Pretty sure only the hammer quality has anything to do with what the report is about, and higher quality would make it *more* likely to break
But of course it's RNG, and a sample size of 3 is basically nothing

**Comment by Eneocho** [comment]
Quality is also dependent on the material (It's why it's recommended to add some gold parts to your iron swords so the higher gold part quality raises the overall construction quality) and phazon is a high quality material. He used phazon, so he got high quality weapons+the buff from the hammer to get the legendaries. Check this PR for some details: #20983

**Comment by Exxion** [comment]
Yes, quality of the resulting tool depends on material quality, but not break chance of the unforged component.

**Comment by Eneocho** [comment]
All I'm saying is that they shouldn't have broken at that point and they didn't. So not a bug.

**Comment by Exxion** [comment]
An unforged component has a chance to break when `strikes` is greater than `strikes_required`. `strikes_required` for a sword blade is 13, but a hammer of higher quality adds more `strikes` per click. I can see from the runtime log that at least one of the swords the OP forged had *60* `strikes`. If my math is right (and the OP is correct that it was 12 clicks), the chance of successfully doing that three times in a row is about 2%, which is not low enough to outright say something is going on, but is worth investigating.

**Comment by Eneocho** [comment]
Fair.
